### PR TITLE
Make transit engine mount path configurable

### PR DESCRIPTION
**Pull request by aaronz-vipaso**
Currently, the mount path of the Transit engine is hard coded to `transit`.  This PR makes it configurable.

### CODE REVIEW

File: runtime/src/main/java/io/quarkus/vault/runtime/client/secretengine/VaultInternalTransitSecretEngine.java [hunk 1]
```diff
@@ -1,83 +1,89 @@
...
 
+    /**
+     * Concatenates the provided URL path segments and sanitizes separators.
+     * The path is returned without a leading "/".
+     *
+     * @param firstPart first URL path segment
+     * @param otherParts other URL path segments
+     * @return String concatenated path
+     */
+    private String getPath(String firstPart, String... otherParts) {
```
Comment (kdubb) [inline runtime/src/main/java/io/quarkus/vault/runtime/client/secretengine/VaultInternalTransitSecretEngine.java:25]: Wouldn't handling nulls here cleanup the call sites even more?

**Comment by kdubb** [comment]
@aaronz-vipaso #184 Adds similar functionality to the KV engine. It also now follows the PKI engine structure.

High level...

1. Add a `mount` to `VaultTransitManager`. Because it pulls double duty as a CDI bean and normal Java instance, you'll need two constructors. One with the mount path configurable, and an `@Inject` constructor that provides the CDI bean with the default mount path.
2. Define a `VaultTransitSecretEngineFactory` interface to vend reactive and regular engines with a provided mount.
3. Add a bean to implement it `VaultTransitManagerFactory`. This will create regular (non-CDI) instances of `VaultTransitManager` using the given mount.

**Comment by kdubb** [comment]
Also, be sure to remove any overloads in the engine interfaces that provide an explicit mount.

File: integration-tests/vault/src/test/java/io/quarkus/vault/VaultITCase.java [hunk 2]
```diff
@@ -30,10 +30,10 @@
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
 
-import java.util.Map;
-import java.util.UUID;
+import java.util.*;
```
[the remark this answers is not in the document]
Comment (aaronz-vipaso) [reply integration-tests/vault/src/test/java/io/quarkus/vault/VaultITCase.java]: done

File: runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java [hunk 3]
```diff
@@ -387,4 +387,15 @@ public Optional<VaultTransitKeyDetail<?>> readKey(String keyName) {
...
     }
+
+    /**
+     * List all Transit keys of the specified Transit engine mount
+     *
+     * @param mount Transit engine mount path
+     * @return key names
+     * @see <a href="https://www.vaultproject.io/api-docs/secret/transit#list-keys">list keys</a>
+     */
+    public List<String> listKeys(String mount) {
```
[the remark this answers is not in the document]
Comment (aaronz-vipaso) [reply runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java]: removed and replaced as requested

File: runtime/src/main/java/io/quarkus/vault/runtime/client/secretengine/VaultInternalTransitSecretEngine.java (hunk 1 above)
Comment (aaronz-vipaso) [reply runtime/src/main/java/io/quarkus/vault/runtime/client/secretengine/VaultInternalTransitSecretEngine.java:38]: In practice, the parameters should not be able to be null. But I have added null handling anyway 👍

**Comment by vashistha** [comment]
I am also looking for transit engine client with mount path support. Please can it be merged and released sooner?

**Comment by kdubb** [comment]
@vashistha @aaronz-vipaso Please look at #214 & #215. It does this and a whole lot more.

If you can please build it locally and comment on #214 with ideas and/or #215 with specific suggestions for changes.

**Comment by aaronz-vipaso** [comment]
> @vashistha @aaronz-vipaso Please look at #214 & #215. It does this and a whole lot more.
> 
> If you can please build it locally and comment on #214 with ideas and/or #215 with specific suggestions for changes.

@kdubb, how long do you expect it will take until the client rewrite is merged? If it takes longer than one month, would it be reasonable to merge this PR in the meantime?

**Comment by kdubb** [comment]
@aaronz-vipaso It's currently quite well tested on it's own (not that there are issues lurking inside). I'm pondering the idea of just releasing a version with the new client and no changes to the existing code.

That would allow people to use it soon and not effect the current stuff until the new client has been proven a reliable replacement.

**Comment by kdubb** [comment]
I already have a `VaultClientProducer` (not checked in) that provides the new client using the current Quarkus configuration. Aside from it being a little confusing (because they are both named `VaultClient` albeit in different packages) it would allow users to inject the new `VaultClient` without any different/extra configuration.

**Comment by kdubb** [comment]
`4.0.0-alpha.2` has been released. Injecting the `VaultClient` (via `@Inject VaultClient client`) and using `client.secrets().transit(mountPath)` allows you to supply any mount path.

I'm closing this now that dynamic mounts should use the client directly instead.

**Pull request closed without merging**